### ISSUE TITLE
Scala 2.12 support

### ISSUE THREAD

**Issue by virtualirfan** [DefinitelyScala/scala-js-template#1]
For certain reasons, I have to use Scala 2.12 on a scala-js project for which I'd really like to use DefinitelyScala scala-js facades. Are you aware of expected problems cross-compiling to Scala 2.12? Thoughts on adding support?

**Comment by virtualirfan** [comment]
fwiw, I tried to move DefinitelyScala/scala-js-auth0 to Scala 2.12 with this diff. At least it compiles so far:
```Scala
index 53c7ead..71bae8b 100644
--- a/project/Projects.scala
+++ b/project/Projects.scala
@@ -11,7 +11,7 @@ object Projects {

   object Versions {
     val app = "1.0.0"
-    val scala = "2.11.8"
+    val scala = "2.12.1"
   }

   private[this] val commonSettings = Seq(
@@ -26,7 +26,8 @@ object Projects {

     shellPrompt := { state => s"[${Project.extract(state).currentProject.id}] $$ " },
     resolvers += Resolver.jcenterRepo,
-    ScalariformKeys.preferences := ScalariformKeys.preferences.value
+    resolvers += Resolver.sonatypeRepo("releases"),
+      ScalariformKeys.preferences := ScalariformKeys.preferences.value
   ) ++ scalariformSettings

   private[this] val scalaJsSettings = Seq(
@@ -45,7 +46,7 @@ object Projects {
     bintrayVcsUrl := Some("git:git@github.com:DefinitelyScala/scala-js-auth0js.git"),
     publishMavenStyle := true,
     licenses += ("MIT", url("http://opensource.org/licenses/MIT")),
-    libraryDependencies ++= Seq("org.scala-js" %%% "scalajs-dom" % "0.8.1"),
+    libraryDependencies ++= Seq("org.scala-js" %%% "scalajs-dom" % "0.9.1"),
     scalaJSStage in Global := FastOptStage
   )
```

**Comment by KyleU** [comment]
Awesome, I'll set them up to cross-publish tonight. Thanks for the initial research. Most of my Scala work is using Play Framework, so I haven't played with 2.12 yet.

**Comment by KyleU** [comment]
I published `auth0` for both 2.11 and 2.12, let me know if it works, and I'll publish the others.

**Comment by KyleU** [comment]
All projects cross-published for 2.11 and 2.12. Thanks for the suggestion!
